### PR TITLE
Rewrite tests using "ftw.testbrowser".

### PR DESCRIPTION
**Pull request by mbaechtold**
This is needed because "ftw.testing" 1.12.0 removed the "splinter" extra.

### CODE REVIEW

File: ftw/publisher/sender/tests/pages.py [hunk 1]
```diff
@@ -1,37 +1,39 @@
...
-            links[self.normalize_whitespace(node.text)] = node
+            links[node.text] = node
 
         assert label in links, 'Could not find transition "%s", got %s' % (
             label, links.keys())
 
         links[label].click()
         if assert_success:
-            self.assert_portal_message('info', 'Item state changed.')
+            assert 'Item state changed.' in statusmessages.info_messages()
```
Comment (jone) [inline ftw/publisher/sender/tests/pages.py]: Use [`statusmessages.assert_message`](http://ftwtestbrowser.readthedocs.io/en/1.23.1/api_pages.html#ftw.testbrowser.pages.statusmessages.assert_message) instead.

File: ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py [hunk 2]
```diff
@@ -1,15 +1,12 @@
-from Products.CMFCore.utils import getToolByName
 from ftw.builder import Builder
 from ftw.builder import create
-from ftw.publisher.sender.testing import PUBLISHER_SENDER_FUNCTIONAL_TESTING
+from ftw.testbrowser.pages import statusmessages
+
```
Comment (jone) [inline ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py]: ✂️

File: ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py [hunk 3]
```diff
@@ -19,112 +16,140 @@
...
+        browser.login().visit(page)
         Workflow().do_transition('submit')
 
-        Workflow().assert_portal_message(
-            'warning', 'The parent object needs to be published first.')
+        self.assertEqual(
+            [
+                'The parent object needs to be published first.'
+            ],
+            statusmessages.warning_messages(),
```
Comment (jone) [inline ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py]: Use [`statusmessages.assert_message`](http://ftwtestbrowser.readthedocs.io/en/1.23.1/api_pages.html#ftw.testbrowser.pages.statusmessages.assert_message) instead.
I'd also assert that there are no error messages:
```python
statusmessages.assert_no_error_messages()
statusmessages.assert_message('The parent object needs to be published first.')
```

File: ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py [hunk 4]
```diff
@@ -19,112 +16,140 @@
...
+        browser.login().visit(page)
         Workflow().do_transition('publish', assert_success=False)
 
-        Workflow().assert_portal_message(
-            'error', 'The parent object needs to be published first.')
+        self.assertEqual(
+            [
+                'The parent object needs to be published first.'
+            ],
+            statusmessages.error_messages(),
```
Comment (jone) [inline ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py]: Use [`statusmessages.assert_message`](http://ftwtestbrowser.readthedocs.io/en/1.23.1/api_pages.html#ftw.testbrowser.pages.statusmessages.assert_message) instead.

File: ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py [hunk 5]
```diff
@@ -19,112 +16,140 @@
...
 
-        Workflow().assert_portal_message(
-            'warning', 'The child object <a href="http://nohost/plone'
-            '/folder/document"></a> is still published.')
+        self.assertEqual(
+            [
+                'The child object <a href="http://nohost/plone'
+                '/folder/document"></a> is still published.'
+            ],
+            statusmessages.warning_messages(),
```
Comment (jone) [inline ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py]: as above

File: ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py [hunk 6]
```diff
@@ -19,112 +16,140 @@
...
```
Comment (jone) [inline ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py]: dito

File: ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py [hunk 7]
```diff
@@ -19,112 +16,140 @@
...
-            'warning', 'The referenced object <a href="http://nohost/plone'
-            '/the-other-page">The Other Page</a> is not yet published.')
-
-    def test_do_not_fail_if_reference_is_none(self):
+        self.assertEqual(
+            [
+                'The referenced object <a href="http://nohost/plone'
+                '/the-other-page">The Other Page</a> is not yet published.'
+            ],
+            statusmessages.warning_messages(),
```
Comment (jone) [inline ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py]: dito

File: ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py [hunk 8]
```diff
@@ -134,28 +159,37 @@ def test_do_not_fail_if_reference_is_none(self):
...
         Workflow().do_transition('publish')
-        self.assertFalse(len(Workflow().portal_messages()['warning']),
-                         'No waring expected, since the ref does not exists '
-                         'anymore.')
-
-    def test_warning_on_publish_when_sl_block_has_unpublished_references(self):
+        self.assertEqual(
+            [],
+            statusmessages.warning_messages(),
+            'No warning expected, since the ref does not exists anymore.'
```
Comment (jone) [inline ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py]: `statusmessages.assert_no_messages` maybe?

File: ftw/publisher/sender/tests/test_publisher_subscribers.py [hunk 9]
```diff
@@ -43,71 +35,69 @@ def test_push_job_in_publisher_queue_after_publishing(self):
...
-    def test_delete_job_create_when_we_have_a_publisher_workflow(self):
+    @browsing
+    def test_delete_job_create_when_we_have_a_publisher_workflow(self, browser):
         queue = IQueue(self.portal)
         folder = create(Builder('folder'))
-        Plone().login().visit(folder, 'delete_confirmation')
+        browser.login().visit(folder, view='delete_confirmation')
         self.assertEquals(0, queue.countJobs())
-        self.click_delete()
+        browser.find('Delete').click()
```
Comment (jone) [inline ftw/publisher/sender/tests/test_publisher_subscribers.py]: You could also use `browser.click_on('Delete')` if you want.

Comment (mbaechtold) [inline ftw/publisher/sender/tests/test_publisher_subscribers.py]: Oh nice, didn't know about that. I have to re-read the docs someday.

**Comment by mbaechtold** [comment]
@jone, I have implemented your suggestions. Could you please have another look?

**Pull request merged**